src/agent/src/rpc.rs

```diff
@@ -1336,6 +1364,23 @@ fn get_memory_info(block_size: bool, hotplug: bool) -> Result<(u64, bool)> {
     Ok((size, plug))
 }
 
+fn get_volume_stats(path: &str) -> Result<(VolumeUsage, VolumeCondition)> {
```

Comment:
This only handles disk usage. What about `inodes` stats if the direct-assigned volume is mounted in the guest? That would be a separate PR?